test/OpenTelemetry.Resources.Azure.Tests/AzureResourceDetectorTests.cs

```diff
@@ -113,7 +113,34 @@ public void AzureContainerAppsResourceDetectorReturnsResourceWithAttributes()
 
         Assert.Contains(new KeyValuePair<string, object>(ResourceSemanticConventions.AttributeServiceName, "containerAppName"), resource.Attributes);
 
-        foreach (var kvp in AzureContainerAppsResourceDetector.AzureContainerResourceAttributes)
+        foreach (var kvp in AzureContainerAppsResourceDetector.AzureContainerAppResourceAttributes)
+        {
+            Assert.Contains(new KeyValuePair<string, object>(kvp.Key, kvp.Key), resource.Attributes);
+        }
+    }
+
+    [Fact]
+    public void AzureContainerAppsJobResourceDetectorReturnsResourceWithAttributes()
+    {
+        try
+        {
+            foreach (var kvp in AzureContainerAppsResourceDetector.AzureContainerAppJobResourceAttributes)
+            {
+                Environment.SetEnvironmentVariable(kvp.Value, kvp.Key);
+            }
+
+            Environment.SetEnvironmentVariable(ResourceAttributeConstants.AzureContainerAppJobNameEnvVar, "containerAppJobName");
+        }
+        catch
+        {
+        }
```

Comment:
I understand that @hansmbakker didn't introduce the try/catch, this was copied from the structure of the above test.

@rajkumar-rangaraj, any idea why this is wrapped in a try/catch? If either of these Sets fail, not only would the test fail but it would be very difficult to identify why.

Reply:
@rajkumar-rangaraj could you please pick this up? I indeed followed the structure from the `AzureContainerAppsResourceDetectorReturnsResourceWithAttributes()` test.

(I cannot finish this)

Reply:
This is not a blocker for merging the PR, so don't worry about this. (Yes, we can follow up separately to address Mothra's comment.)

Reply:
Yes, please don't block on this. I was just asking questions.

Reply:
I resolved the comment